### PR TITLE
feat: handle canary executor response

### PR DESCRIPTION
**Pull request by melanahammel**

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   "private": true,
   "dependencies": {
     "@types/react-flatpickr": "^3.7.1",
+    "@types/react-tooltip": "^3.9.2",
```
Comment (fieldju) [inline packages/client/package.json]: Types should be installed with the -D flag and belong in the devDependencies section.

File: packages/client/src/components/App.tsx [hunk 2]
```diff
@@ -32,9 +32,13 @@ export default class App extends React.Component<AppProps> {
...
                 <Route exact={true} path="/" component={Landing} />
                 <Route path="/docs/:path(.*)" component={Docs} />
-                <Route path="/config/edit/local" component={ConfigEditor} />
-                <Route path="/dev-tools/canary-executor" component={CanaryExecutor} />
-                <Route path="/dev-tools/results" component={CanaryExecutorResults} />
+                <Route path="/config/edit" component={ConfigEditor} />
+                <Route path="/dev-tools/canary-executor/setup" component={CanaryExecutor} />
+                <Route
+                  exact
+                  path="/dev-tools/canary-executor/results\/([0-9A-Z]{26})*"
```
Comment (fieldju) [inline packages/client/src/components/App.tsx]: There is no contract stating that the IDs are 26 chars long, I think `\w+` would be better.

File: packages/client/src/domain/CanaryExecutionRequestTypes.ts [hunk 3]
```diff
@@ -17,5 +17,5 @@ export interface CanaryScope {
   step: number;
   start: string;
   end: string;
-  extendedScopeParams: {};
+  extendedScopeParams: { [key: string]: string };
```
Comment (fieldju) [inline packages/client/src/domain/CanaryExecutionRequestTypes.ts]: Use KvPair type

File: packages/client/src/util/CanaryExecutionFactory.ts [hunk 4]
```diff
@@ -6,22 +6,24 @@ export default class CanaryExecutionFactory {
...
+            scope: 'default',
             location: 'us-west-2',
             step: 60,
-            start: '2019-01-04T18:56:02.816Z',
-            end: '2019-01-04T21:56:02.816Z',
-            extendedScopeParams: {}
+            start: '',
+            end: '',
+            extendedScopeParams: {
+              _scope_key: 'default'
```
Comment (fieldju) [inline packages/client/src/util/CanaryExecutionFactory.ts]: I don't think this should be here, this should default to empty map as most people won't need this.

Comment (fieldju) [reply packages/client/src/util/CanaryExecutionFactory.ts]: Specifically this is specific to the New Relic and SignalFx integrations and in SignalFx this is optional as you can have a default scope set, which we in fact have set for all of our accounts.

File: packages/client/src/util/CanaryExecutionStatusResponseFactory.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import { CanaryExecutionStatusResponse } from '../domain/CanaryExecutionStatusResponse';
+import uuid from 'uuid/v1';
+
+export default class CanaryExecutionStatusResponseFactory {
```
Comment (fieldju) [inline packages/client/src/util/CanaryExecutionStatusResponseFactory.ts]: I don't think you want this file at all, you should be getting the response from the api, and shouldn't have to craft a stub manually.

File: packages/client/src/components/App.tsx (hunk 2 above)
Comment (fieldju) [reply packages/client/src/components/App.tsx]: Also I think you want to do it something like this

`/docs/:path(.*)` where the var is named

File: packages/client/src/domain/CanaryExecutionStatusResponse.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  classification: string;
+  classificationReason: string;
+  id: string;
+  name: string;
+  critical: boolean;
+  groups: [string];
+  controlMetadata: {};
+  experimentMetadata: {};
+  resultMetadata: {};
+  tags: { [key: string]: string };
```
Comment (fieldju) [inline packages/client/src/domain/CanaryExecutionStatusResponse.ts]: You the KvMap here too.

File: packages/client/src/util/CanaryExecutionStatusResponseFactory.ts (hunk 5 above)
Comment (melanahammel) [reply packages/client/src/util/CanaryExecutionStatusResponseFactory.ts]: Refactored and removed this file

File: packages/client/src/domain/CanaryExecutionStatusResponse.ts (hunk 6 above)
Comment (melanahammel) [reply packages/client/src/domain/CanaryExecutionStatusResponse.ts]: implemented

File: packages/client/src/domain/CanaryExecutionRequestTypes.ts (hunk 3 above)
Comment (melanahammel) [reply packages/client/src/domain/CanaryExecutionRequestTypes.ts]: implemented

File: packages/client/src/components/App.tsx (hunk 2 above)
Comment (melanahammel) [reply packages/client/src/components/App.tsx]: implemented

**Pull request merged**